### PR TITLE
fix: implement server-side redirect

### LINKED ISSUE #194: Redirects not working on server side
## Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

## Repro steps.

**Steps:**
1. Configure a redirect in a routes file, e.g. in `src/app/counter/routes.ts`, add `    redirect: '/',`
2. Load `/counter` in the browser

**Actual behavior:**
The server responds with `200 OK` and serves the home page. Then the client will change the path to `/`.

**Expected behavior:**
The server responds with `302 Found` (or `301 Moved Permanently`?) and the client loads `/`.

## Desired functionality.

Behave like a normal web application.

### PR DESCRIPTION
**Pull request by xMartin**
### What is accomplished by your PR?
Fix #194 

### Is there something controversial in your PR?
Not automatically testable as Cypress doesn't support assertion of HTTP status codes nor can you switch off JavaScript in the browser. We could add some tests on the Express level.

### Link to the Issue
#194 

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [ ] Add new passing unit tests to cover the code introduced by your PR
- [ ] Add new documentation for the code introduced by your PR

### CODE REVIEW

File: src/server/isomorphic.ts [hunk 1]
```diff
@@ -95,7 +95,11 @@ export default (context: IServerContext) => {
       }))
              .then(() => {
                context.state = store.state;
```
Comment (xMartin) [inline src/server/isomorphic.ts]: @devCrossNet I wonder what this line does and how it works. I wanted to implement the redirect fix according to the following instructions but couldn't figure out how to pass the router with the context.
https://github.com/vuejs/vue-router/issues/1822#issuecomment-361031675

Comment (devCrossNet) [reply src/server/isomorphic.ts]: I think your code makes much more sense than the one from the example.

the context is the render context of the vue-ssr-renderer and if you provide a state then it automatically creates the __INITIAL_STATE__ variable on the window object.

Basically, the context is everything you can access inside the HTML template.

File: src/server/isomorphic.ts [hunk 2]
```diff
@@ -95,7 +95,11 @@ export default (context: IServerContext) => {
       }))
              .then(() => {
```
Comment (devCrossNet) [inline src/server/isomorphic.ts]: can you format this a little bit different? Can you put `.all` one line down and align it?

```
      Promise
      .all(matchedComponents.map((component: Component) => {
...
      }))
      .then(() => {
...
      })
      .catch(...);
```

**Comment by devCrossNet** [comment]
works as expected 👍 I really like it!

File: src/server/isomorphic.ts (hunk 2 above)
Comment (xMartin) [reply src/server/isomorphic.ts]: I tried to not change the formatting at all but I'm happy to change it to the proposed version.

**Comment by devCrossNet** [comment]
can I rename the commit message to `feat: add server-side redirect`? 

I think a fix would be ok if we already had the functionality before but this is a new feature that I forgot to implement

**Comment by xMartin** [comment]
Sure, you can also view it as a feature.

**Pull request merged**